### PR TITLE
Share implementation of GenerateGuidFromName with EventSourceGenerator.Parser and CoreLib

### PR DESCRIPTION
**Pull request by ericstj**
Originally I had suppressed the CodeQL warning here, but I decided to just share the implementation of this utility method that produces a GUID from name.

### CODE REVIEW

**Comment by ericstj** [comment]
actually, maybe I should just make the implementation equivalent.

File: src/libraries/System.Private.CoreLib/gen/System.Private.CoreLib.Generators.csproj
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/gen/System.Private.CoreLib.Generators.csproj:14]: Is this going to require addition to the exception paperwork for uses of Sha1ForNonSecretPurposes?

cc @GrabYourPitchforks

Comment (ericstj) [reply src/libraries/System.Private.CoreLib/gen/System.Private.CoreLib.Generators.csproj:14]: Initially I had done the suppression and just let it use `Sha1` but this felt more correct.

File: src/libraries/System.Private.CoreLib/gen/System.Private.CoreLib.Generators.csproj
Comment (ericstj) [reply src/libraries/System.Private.CoreLib/gen/System.Private.CoreLib.Generators.csproj:8]: I was able to de-duplicate the source and just link to it, so we no longer have a copy in this project.  I think it's clear what it's for we don't comment on other files added with their reasoning.

**Review by noahfalk — APPROVED** [review]
LGTM
cc @dotnet/dotnet-diag

**Comment by ericstj** [comment]
@noahfalk I might defer to you to merge this if you want to take it.  There was some internal discussion around this approach vs just keeping the code as it is and suppressing the error.  I'd like to make sure the @dotnet/dotnet-diag team makes the final call on this.

**Comment by noahfalk** [comment]
My understanding thus far is that getting a new/expanded exception for using a private implementation of SHA1 (Sha1ForNonSecretPurposes) takes more effort so we are better off not adding another use. Instead we would leave the corelib implementation as a special case and we could eventually get rid of Sha1ForNonSecretPurposes in favor of using a p/invoke to the OS implementation.

Based on that I'm going to close the PR. Feel free to holler if I've misunderstood something.

**Pull request closed without merging**
